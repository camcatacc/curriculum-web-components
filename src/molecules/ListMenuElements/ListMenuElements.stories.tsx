import React, { useState } from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import ListMenuElements from "molecules/ListMenuElements/ListMenuElements";

const Template: ComponentStory<typeof ListMenuElements> = (args) => <ListMenuElements {...args} />;

export const Standard = Template.bind({});

Standard.args = {
	menuElements: [
		{ id: "home", name: "Home", color: "black" },
		{ id: "about", name: "About", color: "black" },
		{ id: "abilities", name: "Abilities", color: "black" }
	],
	selectedId: "about"
};

export default {
	title: "Molecules/ListMenuElements",
	component: ListMenuElements
} as ComponentMeta<typeof ListMenuElements>;
