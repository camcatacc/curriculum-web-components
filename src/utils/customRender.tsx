import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
