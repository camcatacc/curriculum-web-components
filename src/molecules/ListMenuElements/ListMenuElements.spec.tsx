// Modules
import React from "react";
import { fireEvent } from "@testing-library/react";
import { customRender } from "utils/customRender";

// Accessibility
import { axe, toHaveNoViolations } from "jest-axe";

// Elements
import ListMenuElements from "molecules/ListMenuElements/ListMenuElements";

// Definitions
import type { ListMenuElementsProps } from "molecules/ListMenuElements/ListMenuElements";

// Mocks
const onClickMock = jest.fn();

// Default Props
const defaultProps: ListMenuElementsProps = {
	menuElements: [{ name: "Home", onClick: onClickMock, id: "home>" }],
	selectedId: "home"
};

// Tests
expect.extend(toHaveNoViolations);
describe("ListMenuElements", () => {
	beforeEach(() => {
		window.matchMedia = (query) => ({
			matches: false,
			media: query,
			onchange: null,
			addListener: jest.fn(), // Deprecated
			removeListener: jest.fn(), // Deprecated
			addEventListener: jest.fn(),
			removeEventListener: jest.fn(),
			dispatchEvent: jest.fn()
		});
	});

	it("Renders as expected in desktop", async () => {
		const { container } = customRender(<ListMenuElements {...defaultProps} />);
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});

	it("Renders as expected in mobile", async () => {
		window.matchMedia = (query) => ({
			matches: true,
			media: query,
			onchange: null,
			addListener: jest.fn(), // Deprecated
			removeListener: jest.fn(), // Deprecated
			addEventListener: jest.fn(),
			removeEventListener: jest.fn(),
			dispatchEvent: jest.fn()
		});
		const { container } = customRender(<ListMenuElements {...defaultProps} />);
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});

	it("Calls onClick function", () => {
		const { getByText } = customRender(<ListMenuElements {...defaultProps} />);
		const home = getByText("Home");
		fireEvent.click(home);

		expect(onClickMock).toHaveBeenCalled();
	});
});
