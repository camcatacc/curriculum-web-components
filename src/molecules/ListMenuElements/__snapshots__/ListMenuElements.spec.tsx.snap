// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMenuElements Renders as expected in desktop 1`] = `
<div>
  <div
    class="undefined flex-row flex gap-28"
  >
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover sc-bdvvtL clPpEY cursor-pointer font-regular uppercase MuiTypography-colorTextSecondary"
        href="#"
      >
        Home
      </a>
    </p>
  </div>
</div>
`;

exports[`ListMenuElements Renders as expected in mobile 1`] = `<div />`;
